core/services/job/models.go

```diff
@@ -139,7 +139,36 @@ func (s *OffchainReportingOracleSpec) BeforeSave() error {
 	return nil
 }
 
-func (SpecDB) TableName() string                      { return "jobs" }
-func (SpecError) TableName() string                   { return "job_spec_errors_v2" }
-func (OffchainReportingOracleSpec) TableName() string { return "offchainreporting_oracle_specs" }
-func (DirectRequestSpec) TableName() string           { return "direct_request_specs" }
+func (OffchainReportingOracleSpec) TableName() string {
+	return "offchainreporting_oracle_specs"
+}
+
+type DirectRequestSpec struct {
+	IDEmbed
+	ContractAddress models.EIP55Address `json:"contractAddress" toml:"contractAddress"`
+	// OnChainJobSpecID is the sha256 of the TOML that created this job spec
+	OnChainJobSpecID gethCommon.Hash
+	CreatedAt        time.Time `json:"createdAt" toml:"-"`
+	UpdatedAt        time.Time `json:"updatedAt" toml:"-"`
+}
+
+func (DirectRequestSpec) TableName() string {
+	return "direct_request_specs"
+}
+
+type FluxMonitorSpec struct {
```

Comment:
Might be cleaner to have these job type specific structs could go in its respective package?

Reply:
Would it create import cycle problems?

Reply:
shoot yeah it would I just checked.. nvm